hi-ml-azure/src/health_azure/utils.py

```diff
@@ -1252,3 +1255,51 @@ def torch_barrier() -> None:
         return
     if distributed.is_available() and distributed.is_initialized():
         distributed.barrier()
+
+
+def aggregate_hyperdrive_metrics(run_id: str, workspace: Workspace) -> pd.DataFrame:
+    """
+    For a given HyperDrive run id, retrieves the metrics from each of its children and then aggregates it.
+    Returns a DataFrame where each column is one child run, and each row is a metric logged by that child run.
+    For example, for a HyperDrive run with 2 children, where each logs epoch, accuracy and loss, the result
+    would look like:
+
+    |              | 0               | 1                  |
+    |--------------|-----------------|--------------------|
+    | epoch        | [1, 2, 3]       | [1, 2, 3]          |
+    | accuracy     | [0.7, 0.8, 0.9] | [0.71, 0.82, 0.91] |
+    | loss         | [0.5, 0.4, 0.3] | [0.45, 0.37, 0.29] |
+
+    It is possible to log rows and tables in Azure ML by calling run.log_table and run.log_row respectively.
+    In this case, the DataFrame will contain a Dictionary entry instead of a list, where the keys are the
+    table columns (or keywords provided to log_row), and the values are the table values. e.g.
+
+    |                | 0                                        | 1                                         |
+    |----------------|------------------------------------------|-------------------------------------------|
+    | accuracy_table |{'epoch': [1, 2], 'accuracy': [0.7, 0.8]} | {'epoch': [1, 2], 'accuracy': [0.8, 0.9]} |
+
+    It is also posisble to log plots in Azure ML by calling run.log_image and passing in a matplotlib plot. In
+    this case, the DataFrame will contain a string representing the path to the artifact that is generated by AML
+    (the saved plot in the Logs & Outputs pane of your run on the AML portal). E.g.
+
+    |                | 0                                       | 1                                     |
+    |----------------|-----------------------------------------|---------------------------------------|
+    | accuracy_plot  | aml://artifactId/ExperimentRun/dcid.... | aml://artifactId/ExperimentRun/dcid...|
+
+    :param run_id: The run id (type: str) of the parent run. The type of this run must be an AML HyperDriveRun
+    :param workspace: An AzureML Workspace object which contains the run with this id
+    :return: A Pandas DataFrame containing the aggregated metrics from each child run
+    """
+    run = get_aml_run_from_run_id(run_id, aml_workspace=workspace)
+    assert isinstance(run, HyperDriveRun)
+    metrics: DefaultDict = defaultdict()
+    for child_run in run.get_children():
+        child_run_metrics = child_run.get_metrics()
+        child_run_crossval_split = json.loads(child_run.tags.get('hyperparameters')).get('cross_validation_split_index')
```

Comment:
cross_validation_split_index should not be hardcoded

Reply:
Actually - to make that even more generic, can we read out ALL hyperparameters, and attach them as columns to the dataframe?

Reply:
(discussed offline)

Reply:
There is something to be done where still. Your code here will only work if people go with the default argument name. Also, it will fail for any more generic type of hyperdrive run.